monkestation/code/modules/antagonists/mimic/mimic.dm

```diff
@@ -0,0 +1,569 @@
+
+/*
+	AI stuff below here
+*/
+/mob/living/simple_animal/hostile/alien_mimic/CanAttack(atom/the_target)
+	if(the_target == buckled)
+		return TRUE //fixes it jumping off of people immediately
+	if(iscarbon(the_target))
+		var/mob/living/carbon/carbon_target = the_target
+		if(carbon_target.stat == DEAD && should_heal() && !HAS_TRAIT(carbon_target, TRAIT_HUSK))
+			return TRUE
+	if(isliving(the_target))
+		var/mob/living/living_target = the_target
+		var/mob/living/simple_animal/hostile/alien_mimic/attacking_friend = locate() in living_target.buckled_mobs
+		if(attacking_friend && attacking_friend != src)
+			return FALSE
+		var/faction_check = faction_check_mob(living_target)
+		if((faction_check && !attack_same) || living_target.stat)
+			return FALSE
+	return TRUE
+
+/mob/living/simple_animal/hostile/alien_mimic/adjustHealth(amount, updating_health, forced)
+	if(amount > 0 && !mind) //if you take damage, run
+		if(buckled)
+			resist_buckle()
+		if(!target)
+			FindTarget()
+		fleeing = TRUE
+	..()
+
+/mob/living/simple_animal/hostile/alien_mimic/AttackingTarget()
+	if(target == src) //Remove your disguise
+		restore()
+		return
+	if(isliving(target) && !buckled) //Latch onto people
+		var/mob/living/victim = target
+		if(iscarbon(victim) && victim.stat == DEAD && !HAS_TRAIT(victim, TRAIT_HUSK)) //Absorb someone to heal
+			var/mob/living/carbon/carbon_victim = victim
+			if(!carbon_victim.last_mind)
+				to_chat(src, "<span class='warning'>They have no mind to gather information from!</span>")
+				return
+			if(NOHUSK in carbon_victim.dna.species.species_traits)
+				to_chat(src, "<span class='warning'>You can't absorb this person!</span>")
+				return
+			visible_message("<span class='warning'>[src] starts absorbing [carbon_victim]!</span>", \
+						"<span class='userdanger'>You start absorbing [carbon_victim].</span>")
+			if(do_mob(src, carbon_victim, 10 SECONDS))
+				if(HAS_TRAIT(carbon_victim, TRAIT_HUSK)) //Can't let em spam click em
+					return
+				playsound(get_turf(src), absorb_sound,100)
+				visible_message("<span class='warning'>[src] absorbs [carbon_victim]!</span>", \
+							"<span class='userdanger'>[carbon_victim]'s corpse decays as you absorb the nutrients from their body.</span>")
+				carbon_victim.become_husk(MIMIC_ABSORB)
+				people_absorbed++
+				adjustHealth(-40)
+			return
+		if(disguised) //Insta latch if youre disguised
+			if(victim.stat == DEAD)
+				to_chat("<span class='warning'>You can't absorb a body while disguised!</span>")
+				return
+			latch(victim)
+			restore()
+			return
+		else if(do_mob(src, target, 3 SECONDS)) //Latch after a bit if you arent
+			latch(victim)
+			return
+	if(!ismob(target)) //If you're attacking something or other
+		if(disguised)
+			restore()
+		return ..()
+	if(buckled && target == buckled) //If you're buckled to them, or attacking a non-human
+		return ..()
+	if(iscyborg(target) || isAI(target)) //stinky sillicons with their no mounting rules
+		return ..()
```

Comment:
This fires too late, the do_mob still attempts to run before actually attacking.